src/V3EmitCImp.cpp

```diff
@@ -887,19 +899,27 @@ void V3EmitC::emitcImp() {
     UINFO(2, __FUNCTION__ << ": " << endl);
     // Make parent module pointers available.
     const EmitCParentModule emitCParentModule;
+    std::list<std::deque<AstCFile*>> cfiles;
 
     // Process each module in turn
     for (const AstNode* nodep = v3Global.rootp()->modulesp(); nodep; nodep = nodep->nextp()) {
         if (VN_IS(nodep, Class)) continue;  // Imped with ClassPackage
         const AstNodeModule* const modp = VN_AS(nodep, NodeModule);
-        EmitCImp::main(modp, /* slow: */ true);
-        EmitCImp::main(modp, /* slow: */ false);
+        cfiles.push_back(std::deque<AstCFile*>());
+        EmitCImp::main(modp, /* slow: */ true, cfiles.back());
+        cfiles.push_back(std::deque<AstCFile*>());
+        EmitCImp::main(modp, /* slow: */ false, cfiles.back());
     }
 
     // Emit trace routines (currently they can only exist in the top module)
     if (v3Global.opt.trace() && !v3Global.opt.lintOnly()) {
-        EmitCTrace::main(v3Global.rootp()->topModulep(), /* slow: */ true);
-        EmitCTrace::main(v3Global.rootp()->topModulep(), /* slow: */ false);
+        cfiles.push_back(std::deque<AstCFile*>());
+        EmitCTrace::main(v3Global.rootp()->topModulep(), /* slow: */ true, cfiles.back());
+        cfiles.push_back(std::deque<AstCFile*>());
+        EmitCTrace::main(v3Global.rootp()->topModulep(), /* slow: */ false, cfiles.back());
+    }
+    for (auto& l : cfiles) {
+        for (auto cfilep : l) { v3Global.rootp()->addFilesp(cfilep); }
```

Comment:
```suggestion
    for (auto& itr : cfiles) {
        for (auto cfilep : itr) v3Global.rootp()->addFilesp(cfilep);
```

Reply:
FWIW I would go with `coll` for collection or full `itrrable` instead of `itr` (which is ambiguous with iterator which this isn't)

Reply:
"itr" -> iterator reference.  For better or worse this is what is used in much of the other code.

Alternatively can be more specific, e.g. collr - collection reference, or whatever.

Reply:
I changed it to ``collr``.